File: python/2.7/decorator/Mocha.py
from CondimentDecorator import *

class Mocha(CondimentDecorator):
    def __init__(self, beverage):
        self.beverage = beverage

    def getDescription(self):
        return self.beverage.getDescription() + ", Mocha"

    def cost(self):
        return .20 + self.beverage.cost()

File: php/decorator/Mocha.php
<?
include_once("CondimentDecorator.php");

class Mocha extends CondimentDecorator {
    private $beverage;

    public function __construct(Beverage $beverage) {
        $this->beverage = $beverage;
    }

    public function getDescription() {
        return $this->beverage->getDescription() . ", Mocha";
    }

    public function cost() {
        return .20 + $this->beverage->cost();
    }
}
?>